centreon/src/Core/ResourceAccess/Infrastructure/API/UpdateRule/UpdateRuleController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\ResourceAccess\Infrastructure\API\UpdateRule;
+
+use Centreon\Application\Controller\AbstractController;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\ResourceAccess\Application\UseCase\UpdateRule\UpdateRule;
+use Core\ResourceAccess\Application\UseCase\UpdateRule\UpdateRuleRequest;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @phpstan-type _UpdateRequestData array{
+ *     name: string,
+ *     description: string,
+ *     is_enabled: bool,
+ *     contacts: non-empty-list<int>,
+ *     contact_groups: non-empty-list<int>,
+ *     dataset_filters: non-empty-list<array{
+ *      type:string,
+ *      resources: non-empty-list<int>,
+ *      ...
+ *     }>
+ * }
+ */
+final class UpdateRuleController extends AbstractController
+{
+    use LoggerTrait;
+
+    public function __invoke(
+        int $ruleId,
+        Request $request,
+        UpdateRule $useCase,
+        UpdateRulePresenter $presenter
+    ): Response {
+        $this->denyAccessUnlessGrantedForApiConfiguration();
+
+        /** @var _UpdateRequestData $data */
+        $data = $this->validateAndRetrieveDataSent($request, __DIR__ . '/UpdateRuleSchema.yaml');
+        $useCase($this->createDtoFromData($ruleId, $data), $presenter);
+
+        return $presenter->show();
+    }
+
+    /**
+     * @param int $ruleId
+     * @param _UpdateRequestData $data
+     *
+     * @return UpdateRuleRequest
+     */
+    private function createDtoFromData(int $ruleId, array $data): UpdateRuleRequest
+    {
+        $dto = new UpdateRuleRequest();
+        $dto->id = $ruleId;
+        $dto->name = $data['name'];
+        $dto->description = $data['description'];
```

Comment:
```suggestion
        $dto->description = $data['description'] ?? null;
```

In the validation Schema description is not in the required fields, so possibly this array key does not exist

Reply:
Handled differently by setting an empty string instead of null